### PR TITLE
MDEV-32111: Debian Sid does not have libncurses 5.x anymore

### PR DESCRIPTION
**Pull request by illuusio**
- [x] *The Jira issue number for this PR is: MDEV-32111*

## Description
Upstream Debian Sid which will become Debian Trixie (13) have dropped NCruses 5.x and changed dev package name just libncurses-dev

## How can this PR be tested?
Build branch with `debian/autobake-debs.sh` or see branch on Salsa-CI
https://salsa.debian.org/illuusio/mariadb-server/-/pipelines/592558

## Basing the PR against the correct MariaDB version
- [x] *This is a new feature and the PR is based against the latest MariaDB development branch.*

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

File: debian/autobake-deb.sh [hunk 1]
```diff
@@ -94,6 +101,7 @@ case "${LSBNAME}"
 in
   # Debian
   "buster")
+    switch_back_to_ncurses5
```
Comment (grooverdan) [inline debian/autobake-deb.sh:104]: these branches are fall thorough, so only a `bookworm` entry is required.

File: debian/autobake-deb.sh [hunk 2]
```diff
@@ -122,17 +132,21 @@ in
     ;;
   # Ubuntu
   "bionic")
+    switch_back_to_ncurses5
     remove_rocksdb_tools
     [ "$architecture" != amd64 ] && disable_pmem
     ;&
   "focal")
+    switch_back_to_ncurses5
```
Comment (grooverdan) [inline debian/autobake-deb.sh:140]: these fall though too.

Why is Ubuntu being forced back to ncurses5 when it has ncurses (6.x+)? We'll need to manually add them to bb however (like https://github.com/MariaDB/buildbot/pull/168).

File: debian/autobake-deb.sh (hunk 1 above)
Comment (illuusio) [reply debian/autobake-deb.sh:104]: Good point I'll sorted that out in next version

File: debian/autobake-deb.sh (hunk 2 above)
Comment (illuusio) [reply debian/autobake-deb.sh:140]: Ain't they are fetch on build time? I'll  make Pull for BB also.

Comment (illuusio) [reply debian/autobake-deb.sh]: It seems only Buster (which I'll make remove PR soon) should have Ncurses 5 others have Ncurses 6.2

File: debian/autobake-deb.sh (hunk 1 above)
Comment (grooverdan) [reply debian/autobake-deb.sh:104]: and buster has libbcurses-dev as 6 - https://packages.debian.org/buster/libncurses-dev.

With bionic at EOL the switchback function can be removed along with the bionic case statement.

Lets rebase to 11.4 too.

Comment (illuusio) [reply debian/autobake-deb.sh:104]: I've rebased and I'll remove it. Should Bionic be removed all along from from 10.4 or is there another build coming up?

Comment (grooverdan) [reply debian/autobake-deb.sh:104]: Sorry I did typo 11.4 meaning 10.4. Release is due out any day now, so a rebase and we'll just merge it after the release is out.

Comment (illuusio) [reply debian/autobake-deb.sh:104]: Ok this should be rebased to 10.4 and remove from there. Ok I'll do it.

**Comment by Justman100** [comment]
What is here now? Can no longer install libncurses5, which MariaDB still needs

**Comment by illuusio** [comment]
> What is here now? Can no longer install libncurses5, which MariaDB still needs

Sorry I didn't quite get the point. MariaDB works fine with NCurses 6.x and Trixie has marked NCurses obsoleted.

**Comment by Justman100** [comment]
> > What is here now? Can no longer install libncurses5, which MariaDB still needs
> 
> Sorry I didn't quite get the point. MariaDB works fine with NCurses 6.x and Trixie has marked NCurses obsoleted.

Yes, already seen at [packages.debian.org](https://packages.debian.org/search?keywords=libncurses5)

**Comment by Justman100** [comment]
> > What is here now? Can no longer install libncurses5, which MariaDB still needs
> 
> Sorry I didn't quite get the point. MariaDB works fine with NCurses 6.x and Trixie has marked NCurses obsoleted.

I have checked again! Installing and starting MariaDB works! However, if you now want to go to the console:

> mariadb: error while loading shared libraries: libncurses.so.5: cannot open shared object file: No such file or directory

**Comment by Justman100** [comment]
@dr-m

Can you merge this? It's very important!!!

**Comment by Justman100** [comment]
@grooverdan

Why in 10.5? The latest version is 11.4! LOOOLLLL???

**Comment by grooverdan** [comment]
because changes are merged from earliest maintained version to latest verion.

**Pull request merged**